astropy/time/tests/test_precision.py

```diff
@@ -198,3 +199,12 @@ def test_conversion_preserves_jd1_jd2_invariant_2():
     assert t2.jd1 % 1 == 0
     assert abs(t2.jd2) <= 0.5
     assert abs(t2.jd2) < 0.5 or t2.jd1 % 2 == 0
+
+
+def test_datetime_difference_agrees_with_timedelta_no_hypothesis():
```

Comment:
Nitpick, but omit `_no_hypothesis()` (at the very least for this PR!).

Reply:
It's all a bit awkward because I don't trust this to be a useful test; it's the hypothesis version that is the real test, and I wouldn't have recommended merging this until it passed the hypothesis testing. So this clone is largely decorative.

Reply:
Don't quite disagree - this was a test that showed the previous implementation was wrong, so it is good to guard against regression regardless!

Reply:
Well, but I think it takes the hypothesis tests to be confident of avoiding regression. And they exist, maintained in parallel to this one.